```
======================================================================
ERROR: test_edit_submissions_simple (corehq.apps.commtrack.tests.test_rebuild.RebuildStockStateTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 232, in inner
    return helper(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 205, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 220, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 201, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/tests/test_rebuild.py", line 123, in test_edit_submissions_simple
    form_id=form.form_id,
  File "/mnt/commcare-hq-ro/corehq/apps/hqcase/utils.py", line 54, in submit_case_blocks
    attachments=attachments,
  File "/mnt/commcare-hq-ro/corehq/apps/receiverwrapper/util.py", line 29, in submit_form_locally
    **kwargs
  File "/mnt/commcare-hq-ro/corehq/form_processor/submission_post.py", line 144, in run
    result = process_xform_xml(self.domain, self.instance, self.attachments)
  File "/mnt/commcare-hq-ro/corehq/form_processor/parsers/form.py", line 81, in process_xform_xml
    return _handle_id_conflict(instance, e.xform, domain)
  File "/mnt/commcare-hq-ro/corehq/form_processor/parsers/form.py", line 151, in _handle_id_conflict
    return _handle_duplicate(xform, instance)
  File "/mnt/commcare-hq-ro/corehq/form_processor/parsers/form.py", line 180, in _handle_duplicate
    existing_doc, new_doc = apply_deprecation(existing_doc, new_doc, interface)
  File "/mnt/commcare-hq-ro/corehq/form_processor/parsers/form.py", line 198, in apply_deprecation
    if existing_xform.persistent_blobs:
AttributeError: 'XFormInstanceSQL' object has no attribute 'persistent_blobs'
-------------------- >> begin captured stdout << ---------------------
<function test_edit_submissions_simple at 0x7f92051027d0> failed with the following settings:
settings.TESTS_SHOULD_USE_SQL_BACKEND = True
```

--- a/corehq/form_processor/models.py
+++ b/corehq/form_processor/models.py
@@ -219,6 +219,9 @@ class XFormInstanceSQL(DisabledDbMixin, models.Model, RedisLockableMixIn, Attach
     deleted_on = models.DateTimeField(null=True)
     deletion_id = models.CharField(max_length=255, null=True)
 
+    # for compatability with corehq.blobs.mixin.DeferredBlobMixin interface
+    persistent_blobs = None
+
     @classmethod
     def get_obj_id(cls, obj):
         return obj.form_id

--- a/corehq/form_processor/parsers/form.py
+++ b/corehq/form_processor/parsers/form.py
@@ -198,7 +198,7 @@ def apply_deprecation(existing_xform, new_xform, interface=None):
     if existing_xform.persistent_blobs:
         for name, meta in existing_xform.persistent_blobs.items():
             with existing_xform.fetch_attachment(name, stream=True) as content:
-                new_xform.deferred_put_attachment(
+                existing_xform.deferred_put_attachment(
                     content,
                     name=name,
                     content_type=meta.content_type,

